ui/ui-components/client/client.ts

```diff
@@ -83,92 +89,91 @@ export class Client {
   private serverToken = process.env.SERVER_TOKEN;
   private cache = new ClientCache();
 
-  private static readonly optionDefaults: ClientRequestOptions = {
+  private static readonly defaultOptions: ClientRequestOptions = {
     useCache: true,
-    errorHandler,
+    errorHandler: isBrowser() ? errorHandler : undefined,
+  };
+
+  private static readonly defaultHeaders: AxiosRequestHeaders = {
+    Accept: "application/json",
+    "Content-Type": "application/json",
+    "X-Grouparoo-Client": `${PackageJSON.name}-v${PackageJSON.version}`,
   };
 
   constructor(
     private getRequestContext: () => {
+      type?: NextContextType;
       req?: IncomingMessage;
       res?: ServerResponse;
     } = () => ({})
   ) {}
 
-  private csrfToken = () => {
-    if (globalThis?.localStorage) {
-      return window.localStorage.getItem("session:csrfToken");
-    }
-  };
+  private redirectOnAccessError({ code }: { code: string }): boolean {
```

Comment:
I think I'm still confused to why this is needed within the client vs the users of the client.  My understanding is that there are only 2 callers of the client: in-browser react callers or server-side hydration callers

* In all cases of the in-browser react callers, an errorHandler will be present and take care of problems. The errorHandler may want to have some redirect logic, but the client doesn't need it
* In the cases of the server-side hydrators, either
  1. `getInitialProps` - throwing will redirect the user to the `_error` page where we provide a "sign in" button for AUTH* error codes.  We are deprecating `getInitialProps`, so this can go away soon
  2. `getServerSideProps` - we wrap all uses of `getServerSideProps` with `withServerErrorHandler` which handles the redirect

Reply:
I think there's a case where the react renderer on the server may make requests, but I'm not 100% on that.